### DISCUSSION TITLE
how to extract addresses using spacy and NER and say whether it is real or fake

### DISCUSSION THREAD

**Discussion by kranthiksk** [explosion/spaCy#11027] [Help: Coding & Implementations] [answered]
![image](https://user-images.githubusercontent.com/98621657/175655581-1a2bdcda-e74e-4519-b6ff-527d605a8c81.png)
How do I detect addresses using NER and spaCy. NER detects names, organizations, dates, book titles, etc. While there are a lot of pre-trained models for names, there are none for addresses.

In the same way is it possible to detect addresses, street names, cities, zip codes, etc. Also, after detecting the address, is it possible to say whether it is Real or Fake?

Also, I find it difficult to get the datasets. Kindly provide me with your suggestions or the best approach.

Trying to get a similar visualization for the addresses also.

**Comment by pmbaumgartner** [comment, marked as answer]
Hey @kranthiksk - thanks for the question.

There are some resources on this question here: https://github.com/explosion/spaCy/issues/1655. I'd recommend checking out the tools and methods that are referenced in that discussion. 

For detecting a "real or fake" address, aka validation, I'd search for an API that offers that as it'll depend on what country your addresses are from.

**Reply by kranthiksk** [reply]
Thank you @pmbaumgartner. I will study the resources which you have mentioned. If I have any difficulty, will get back.
